google/resource_compute_backend_service_migrate.go

```diff
@@ -0,0 +1,94 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func resourceComputeBackendServiceMigrateState(
+	v int, is *terraform.InstanceState, meta interface{}) (*terraform.InstanceState, error) {
+	if is.Empty() {
+		log.Println("[DEBUG] Empty InstanceState; nothing to migrate.")
+		return is, nil
+	}
+
+	switch v {
+	case 0:
+		log.Println("[INFO] Found Compute Backend Service State v0; migrating to v1")
+		is, err := migrateBackendServiceStateV0toV1(is)
+		if err != nil {
+			return is, err
+		}
+		return is, nil
+	default:
+		return is, fmt.Errorf("Unexpected schema version: %d", v)
+	}
+}
+
+func migrateBackendServiceStateV0toV1(is *terraform.InstanceState) (*terraform.InstanceState, error) {
+	log.Printf("[DEBUG] Attributes before migration: %#v", is.Attributes)
+
+	oldHashToValue := map[string]map[string]interface{}{}
+	for k, v := range is.Attributes {
+		if !strings.HasPrefix(k, "backend.") {
+			continue
+		}
+
+		if k == "backend.#" {
```

Comment:
You could move it in the if above with a `||`.

Reply:
Done